### DISCUSSION TITLE
Builder package

### DISCUSSION THREAD

**Discussion by audrenbdb** [aarondl/sqlboiler#1393] [Ideas]
Hey, 

At our company we are using sqlboiler a lot and it is great.

For our tests, we often needed to generate random struct with random populated data.

Code size growing, and amount of tables as well, I made a code generator that run after the sqlboiler generator, to create a struct with random populated properties. 

It is using the builder pattern. 

Syntax example :
builder.NewProduct().WithQty(5)

Is it something that would be relevant to propose for the sqlboiler project ? Has such thing been discussed before ?

I'd be happy to contribute should this be desirable.

**Comment by stephenafamo** [comment]
Looks like what you're looking for is [boilingfactory](https://github.com/stephenafamo/boilingfactory)

I created this cause I needed this too. There may be some rough edges, it's been a while since I used it. 
It should be possible to include this directly in SQLBoiler, because the meat of it is some additional templates. 

EDIT: It seems you have your own version of a factory, which also includes randomizing fields by default.  
That is pretty nice! It may be better than boilingfactory, or perhaps there are things to combine from both projects.
